New example for a data playground Terraform setup

aymanfarhat opened the pull request:
A simple template for initialising a new project to be used as a data playground / data exploration project with the basics in place.

ludoo commented on examples/data-solutions/data-playground/variables.tf:
```diff
@@ -0,0 +1,64 @@
+  default     = "europe"
+}
+
+variable "project_name" {
```
We use `project_id` across Fabric, can you rename this and also:

- allow reusing an existing project
- add a `prefix` variable

aymanfarhat replied:
Done.

ludoo commented on examples/data-solutions/data-playground/variables.tf:
```diff
@@ -0,0 +1,64 @@
+  default     = "europe-west1"
+}
+
+variable "zone" {
```
no need for a variable for this, you can either

- have only a zone variable and derive the region from that, or
- use a fixed `b` for zone in code, users can easily change it

aymanfarhat replied:
Done.

ludoo commented on examples/data-solutions/data-playground/variables.tf:
```diff
@@ -0,0 +1,64 @@
+  type        = string
+}
+
+variable "vpc_ip_cidr_range" {
```
can you bundle all VPC related variables in a single `vpc_config` variable of type object?

aymanfarhat replied:
Done

ludoo commented:
THis is great, thanks for the changes. As a last step, can you add a simple test like for the other examples, in the `tests/examples` folder?

aymanfarhat commented:
@ludoo Thanks for the feedback! I added a test folder as requested and addressed the comments mentioned by @lcaggio as well (reviewed them in a 1:1).

ludoo commented:
Thanks for adding tests and implementing changes! The only things left before this can be merged are

- a simple diagram that is inline with the style we use in the other examples
- alphabetical ordering of variables :)

aymanfarhat commented:
Hello! I hope all is well. Just checking-in regarding this PR, is there anything else needed from my end? I see that it's approved but seems to be pending a merge. Thanks!
